spec/requests/api/v2/providers/courses/sync_with_search_and_compare_spec.rb

```diff
@@ -84,5 +81,37 @@
       let(:manage_api_response) { '{ "result": true }' }
       it { should have_http_status(:internal_server_error) }
     end
+
+    describe 'failed validation' do
+      let(:json_data) { JSON.parse(subject.body)['errors'] }
+
+      context 'no enrichments' do
+        let(:course) { create(:course, provider: provider, with_enrichments: []) }
+        it { should have_http_status(:unprocessable_entity) }
+        it 'has validation errors' do
+          expect(json_data.count).to eq 1
+          expect(response.body).to include('Invalid enrichment')
+          expect(response.body).to include("Enrichments can't be blank")
+        end
+      end
+
+      context 'invalid enrichment' do
+        let(:invalid_enrichment) { create(:course_enrichment, :invalid_content) }
+
+        let(:course) { create(:course, :fee_type_based, provider: provider, enrichments: [invalid_enrichment]) }
+        it { should have_http_status(:unprocessable_entity) }
+
+        it 'has validation errors' do
+          expect(json_data.count).to eq 6
+          expect(response.body).to include('Invalid latest_enrichment')
+          expect(response.body).to include("Latest enrichment About course Reduce the word count for about course")
```

Comment:
Hmm, I would be interested if we can clean these up and remove the humanised field name at the start, `About course Reduce ...` -> `Reduce ...`

Reply:
https://stackoverflow.com/a/40254247

Reply:
Okay, I'm happy to do this as a follow-up, will card it up. :)

Reply:
This too: https://stackoverflow.com/a/43565674

Reply:
just fyi this test is just a little bit misleading, as I was clueless to what we should have it return

the `json_data`/`response.body` is structured like 

```ruby[{
    "title" => "Invalid latest_enrichment",
    "detail" =>
    "Latest enrichment About course Reduce the word count for about course",
    "source" => {}
  },
  {
    "title" => "Invalid latest_enrichment",
    "detail" =>
    "Latest enrichment Interview process Reduce the word count for interview process",
    "source" => {}
  },
  {
    "title" => "Invalid latest_enrichment",
    "detail" =>
    "Latest enrichment How school placements work Reduce the word count for how school placements work",
    "source" => {}
  },
  {
    "title" => "Invalid latest_enrichment",
    "detail" =>
    "Latest enrichment Fee international must be greater than or equal to 0",
    "source" => {}
  },
  {
    "title" => "Invalid latest_enrichment",
    "detail" => "Latest enrichment Fee uk eu must be less than or equal to 100000",
    "source" => {}
  },
  {
    "title" => "Invalid latest_enrichment",
    "detail" =>
    "Latest enrichment Fee details Reduce the word count for fee details",
    "source" => {}
]
```

Reply:
It's okay I think